File: cs50/week1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void brick(int w, int h)
{
    for (int i = 0; i < h - w; i++)//w段目はh-wだけ空白出力
    {
        printf(" ");
    }

    for (int i = 0; i < w; i++)//残ったwだけ#出力
    {
        printf("#");
    }

    printf("\n");
}

int main(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8); //条件に合うまで入力を取得

    for (int i = 1; i <= n; i++)//各行についてbrick
    {
        brick(i, n);
    }
}

File: cs50/week6/sentimental-mario-less/mario.py
import cs50
# TODO
h = 0
while (h < 1) | (h > 8):  # 大きさが制限外なら聞き続ける
    h = cs50.get_int("Height: ")

for i in range(h):
    s = ''  # 行の変数を用意、初期化
    for j in range(h - i - 1):  # 空白の出力
        s += " "
    for k in range(i + 1):  # "#"の出力
        s += "#"
    print(s)  # 表示